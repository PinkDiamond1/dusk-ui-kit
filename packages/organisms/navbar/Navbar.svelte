<script>
  import { setContext } from "svelte";
  import { current_component } from "svelte/internal";
  import {
    forwardEventsBuilder,
    exclude,
    useActions,
    contexts,
    variants,
  } from "@dusk-network/helpers";
  import Icon from "@dusk-network/icon";
  import Logo from "@dusk-network/logo";
  import AppsMenu from "./menu/AppsMenu.svelte";
  import LinksMenu from "./menu/LinksMenu.svelte";
  import "./styles.css";
  const forwardEvents = forwardEventsBuilder(current_component);

  export let use = [];
  export let title;
  export let apps;
  export let links;
  export let appName;

  let className = "";
  export { className as class };

  let expanded = false;

  export let id;
  export let variant = variants.ORGANISM.NAVBAR.BRAND;
  setContext("DUK:menu:context", contexts.MENU.NAVBAR);
  setContext("DUK:logo:context", contexts.LOGO.NAVBAR);

  function getClassNames(variant) {
    let classNames = "";
    switch (variant) {
      case variants.ORGANISM.NAVBAR.BRAND:
        classNames += " duk-navbar--brand";
        break;
      case variants.ORGANISM.NAVBAR.CTA:
        classNames += " duk-navbar--cta";
        break;
      case variants.ORGANISM.NAVBAR.INFO:
        classNames += " duk-navbar--info";
        break;
      case variants.ORGANISM.NAVBAR.SUCCESS:
        classNames += " duk-navbar--success";
        break;
      case variants.ORGANISM.NAVBAR.WARNING:
        classNames += " duk-navbar--warning";
        break;
      case variants.ORGANISM.NAVBAR.DANGER:
        classNames += " duk-navbar--danger";
        break;
      case variants.ORGANISM.NAVBAR.LIGHT:
        classNames += " duk-navbar--light";
        break;
      case variants.ORGANISM.NAVBAR.DARK:
        classNames += " duk-navbar--dark";
        break;
      case variants.ORGANISM.NAVBAR.BLACK:
        classNames += " duk-navbar--black";
        break;
      case variants.ORGANISM.NAVBAR.WHITE:
        classNames += " duk-navbar--white";
        break;
      default:
        classNames += "";
    }

    return classNames;
  }

  const toggleNavbar = () => {
    // console.log(id);
    // console.log(document.getElementById(`#${id}`));
    const navbarElement = document.querySelector(`#${id}`);
    navbarElement.classList.toggle("duk-navbar--hidden");

    expanded = !navbarElement.classList.contains("duk-navbar--hidden");
  };
</script>

<nav
  use:useActions="{use}"
  use:forwardEvents
  class="duk-navbar {className} duk-navbar--hidden
  {getClassNames(variant)}"
  {...exclude($$props, ['use', 'class', 'variant', 'title', 'apps', 'links', 'appName'])}>
  <div class="duk-navbar__wrapper duk-navbar__wrapper--primary">
    <div class="duk-navbar__brand">
      {#if $$slots.logo}
        <slot name="logo" />
      {:else}
        <Logo class="duk-navbar__logo" href="https://dusk.network" />
      {/if}
      {#if title}
        <h1 class="duk-navbar__brand__heading">{title}</h1>
      {/if}
      <div class="duk-navbar__navigation duk-navbar__navigation--primary">
        {#if $$slots.networks}
          <slot name="networks" />
        {/if}
        {#if $$slots.apps}
          <slot name="apps" />
        {:else}
          <AppsMenu apps="{apps}" appName="{appName}" />
        {/if}
      </div>
    </div>
    <div class="duk-navbar__navigation duk-navbar__navigation--tertiary">
      {#if $$slots.links}
        <slot name="links" />
      {:else}
        <LinksMenu links="{links}" />
      {/if}
    </div>
    <div id="{id}-content" class="duk-navbar__collapse">
      <button
        id="__DUK-navbar-toggle"
        aria-controls="{id}-content"
        aria-expanded="{expanded}"
        aria-label="Toggle navigation"
        on:click="{toggleNavbar}"
        class="duk-navbar__collapse__button">
        {#if !expanded}
          <Icon name="menu-burger" />
        {:else}
          <Icon name="menu-burger-close" />
        {/if}
      </button>
    </div>
  </div>
  <div class="duk-navbar__wrapper duk-navbar__wrapper--secondary">
    <div class="duk-navbar__navigation duk-navbar__navigation--secondary">
      <slot name="navigation" />
    </div>
  </div>
</nav>
