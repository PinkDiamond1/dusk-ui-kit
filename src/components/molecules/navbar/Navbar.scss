.__duk-navbar {
  .nav-link {
    font-weight: 300;
    &:last-child {
      padding-bottom: 0;
      padding-right: 0;
    }
  }
}
