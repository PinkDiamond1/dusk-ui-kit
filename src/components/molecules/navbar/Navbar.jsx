/* @flow */
import * as React from 'react'
import BootstrapNav from 'react-bootstrap/Nav'

import NavbarItem from '../../atoms/navbar-item'

import './Navbar.scss'

type Props = {
  children: React.Node,
  className: string,
}

const Navbar = (props: Props): React.Element<*> => {
  const { children, className } = props
  const updatedClassName = `__duk-navbar ${className}`
  return <BootstrapNav className={updatedClassName}>{children}</BootstrapNav>
}

Navbar.defaultProps = {
  className: '',
}

Navbar.Item = NavbarItem

export default Navbar
