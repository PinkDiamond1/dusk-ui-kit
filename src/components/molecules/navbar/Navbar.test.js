import React from 'react'
import renderer from 'react-test-renderer'

import Navbar from './Navbar'

it('test to see if the Navbar renders correctly', () => {
  const tree = renderer.create(<Navbar>Smoke Test</Navbar>).toJSON()
  expect(tree).toMatchSnapshot()
})
