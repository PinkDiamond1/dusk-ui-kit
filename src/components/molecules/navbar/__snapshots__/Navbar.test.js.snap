// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the Navbar renders correctly 1`] = `
<div
  className="__duk-navbar  nav"
  onKeyDown={[Function]}
>
  Smoke Test
</div>
`;
